### PR TITLE
Tap tester

### PR DESCRIPTION
**Pull request by pbegle**
# Description of change
Adding tap tester.

# Manual QA steps
 - 
 
# Risks
 - I don't really have the ability to run this so not sure that it actually works.
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import os
+import unittest
+
+from tap_sailthru.discover import get_schemas
```
Comment (kspeer825) [inline tests/base.py:4]: Dependencies on the tap code should be avoided at all costs in these tests. It is important that expectations are set separately based on the business logic of the 3rd party application and product requirements, not based on the implementation of those requirements in the tap.

File: tests/base.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    def untestable_streams(self):
+        return set()
+
+    def expected_sync_streams(self):
+        return self.expected_streams() - self.untestable_streams()
+
+    def expected_metadata(self):
+        _, metadata = get_schemas()
+        return {table: metadata[table][0]['metadata'] for table in metadata.keys()}
```
Comment (kspeer825) [inline tests/base.py:45]: This metadata should be explicitly written out using the constants at the top of the class, not set based off of the tap implementation.

File: tests/base.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        # Assert that the check job succeeded
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+    def get_properties(self, original_properties=True):
+        properties = {
+            'start_date': '2017-01-01T00:00:00Z',
+            'user_agent': 'tap-sailthru tap-tester@test.com',
+            'api_key': os.environ['TAP_SAILTHRU_API_KEY'],
+            'api_secret': os.environ['TAP_SAILTHRU_API_SECRET'],
```
Comment (kspeer825) [inline tests/base.py:65]: I believe the key-value pairs set in `get_properties` and `get_credentials` should be mutually exclusive. They should be dropped here if they are in fact credentials.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

**Comment by pbegle** [comment]
@kspeer825 I made the changes you requested. Let me know if that looks good now or if there are additional changes.

**Review by kspeer825 — APPROVED** [review]
Thanks @pbegle, looks good.

**Pull request merged**